mirrord/cli/src/main.rs

```diff
@@ -236,11 +208,39 @@ fn exec(args: &ExecArgs) -> Result<()> {
     let library_path = extract_library(args.extract_path.clone())?;
     add_to_preload(library_path.to_str().unwrap()).unwrap();
 
+    #[cfg(target_os = "macos")]
```

Comment:
Thinking out loud - maybe we want to check if rosetta is available before doing the patch?
Also the desired behavior might be just a warning and then "not loading" to the said binary?
wdyt?

Reply:
> just a warning and then "not loading" to the said binary

Do you mean letting the application run without mirrord?

Reply:
yeah, on your next feature that might be the situation when exec and it might not affect the work"iness"

Reply:
Yeah on an `exec` hook it might make sense to keep running even if the patch failed, but here it's the main binary the user is trying to run, so if we can't load into there we're just not running at all. I think just running the binary without loading into it is not helpful for the user, as they can also do that without mirrord, and it's possible that they don't notice the warning and think mirrord is running and just don't understand why it's not working as they expect.

Reply:
yes though there might be situations where we are wrong - we detected SIP but it ain't really protected.

Reply:
I think the current flow is good:
Patch if SIP -->  run --> if failed because of rosetta show explanation

But should I add rosetta detection and change to:
Patch if SIP and x86/has rosetta --> run (even if there's SIP but we didn't patch for lack of rosetta).
? 

I don't know how reliably we can detect rosetta.

One other possible flow: 
Patch if SIP --> run --> if failed to run patched, emit error, then try to run original binary.
In this flow, like in the second one, we would probably sometimes be running just the plain binary without the layer being loaded, which I think we should not. But we would not have false negatives for rosetta because we would just do the patch and try to run it.

All in all I think the first flow is the best one, but let me know which one you like best.

Reply:
okay you got me. Let's see how users react.